onnxruntime/core/session/inference_session.cc

```diff
@@ -156,18 +156,14 @@ class InferenceSession::Impl {
 
     InitLogger(logging_manager);
 
-    // currently the threadpool is used by the parallel executor only and hence
-    // there is no point creating it when only sequential execution is enabled.
-    if (!session_options.enable_sequential_execution) {
-      int pool_size = session_options_.session_thread_pool_size == 0
+    // The threadpool is currently evolving.  We will always create a per session threadpool.
+    // Beyond this, we will create a global thread pool to share across sessions.
```

Comment:
>global thread pool [](start = 37, length = 18)

Would this be some sort of environment level thread pool?

Reply:
Yes.  Handling multiple concurrent sessions while avoiding over allocation of threads requires this.